packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -160,7 +160,7 @@ export class WebSocketConnectionProvider {
      */
     protected createWebSocketUrl(path: string): string {
         const endpoint = new Endpoint({ path });
-        return endpoint.getWebSocketUrl().toString();
+        return endpoint.getWebSocketUrl().toString(true);
```

Comment:
Is it really a good idea to disable encoding of URLs?

Reply:
Would it be better instead to change the spec?

https://github.com/theia-ide/theia/blob/master/packages/core/src/node/messaging/messaging-contribution.ts#L196

So that it would also match an encoded version of the URL?

Reply:
So you mean `url.parse` does not decode `path` here properly: https://github.com/theia-ide/theia/blob/3b0d18071dfc1a72cb42807075bb3fb36d63375a/packages/core/src/node/messaging/messaging-contribution.ts#L111 ?

We have to check it, but it would be good to get an URL first to know what to check.

Reply:
To be fair, I can see it break in different places, but apparently the URL can look like `/users/rayh@example.com/theia`.

Although it can break before even getting to Theia: @rayh are you using some kind of reverse proxy? If so, are you able to tell if it is failing at the proxy level rather than Theia?

If it is coming from Theia, it would help to try and debug from the point where the ws handshake is received and the handling logic happens, to see where it fails exactly. See the link sent by Anton.